src/mx_bluesky/common/external_interaction/callbacks/xray_centre/ispyb_callback.py

```diff
@@ -281,5 +285,10 @@ def activity_gated_stop(self, doc: RunStop) -> RunStop:
             )
             if self.ispyb_ids == IspybIds():
                 raise ISPyBDepositionNotMade("ispyb was not initialised at run start")
+            exception_type, message = SampleException.type_and_message_from_reason(
+                doc.get("reason", "")
+            )
+            if exception_type == CrystalNotFoundException.__name__:
+                doc["reason"] = message
```

Comment:
Could: I think we can just do this for all exceptions e.g.
```suggestion
            if exception_type:
                doc["reason"] = message
```

Reply:
Ah yeah, I think this is much better; no point restricting what errors we append reasons for.